implementStr.java

```diff
@@ -0,0 +1,16 @@
+//tc : o(n)
```

Comment:
TC: O(N*M) for the written solution. Please write the optimized solutions as well.